options/options_type.h

```diff
@@ -367,18 +439,56 @@ class OptionTypeInfo {
                      const std::string& opt_name, const char* this_ptr,
                      const std::string& that_value) const;
 
+  // Parses the input value according to the map for the struct at opt_addr
+  // struct_name is the name of the struct option as registered
+  // opt_name is the name of the option being evaluated.  This may
+  // be the whole struct or a sub-element of it, based on struct_name and
+  // opt_name.
```

Comment:
It would be better to give an example to explain "This maybe the whole struct or a sub-element of it"